tests/components/forked_daapd/test_config_flow.py

```diff
@@ -0,0 +1,181 @@
+        autospec=True,
+    ) as mock_get_request:
+        mock_get_request.return_value = SAMPLE_CONFIG
+        mock_test_connection.return_value = ["ok", "My Music on myhost"]
+        config_data = config_entry.data
+        result = await hass.config_entries.flow.async_init(
+            DOMAIN, context={"source": SOURCE_USER}, data=config_data
+        )
+        await hass.async_block_till_done()
+        assert result["type"] == data_entry_flow.RESULT_TYPE_CREATE_ENTRY
+        assert result["title"] == "My Music on myhost"
+        assert result["data"][CONF_HOST] == config_data[CONF_HOST]
+        assert result["data"][CONF_PORT] == config_data[CONF_PORT]
+        assert result["data"][CONF_PASSWORD] == config_data[CONF_PASSWORD]
+
+        # Also test that creating a new entry with the same host aborts
+        result = await hass.config_entries.flow.async_init(
+            DOMAIN, context={"source": SOURCE_USER}, data=config_entry.data,
+        )
+        await hass.async_block_till_done()
+        assert result["type"] == data_entry_flow.RESULT_TYPE_ABORT
+
+
+async def test_zeroconf_updates_title(hass, config_entry):
+    """Test that zeroconf updates title and aborts with same host."""
+    MockConfigEntry(domain=DOMAIN, data={CONF_HOST: "different host"}).add_to_hass(hass)
+    config_entry.add_to_hass(hass)
+    assert len(hass.config_entries.async_entries(DOMAIN)) == 2
+    discovery_info = {
+        "host": "192.168.1.1",
+        "port": 23,
+        "properties": {"mtd-version": 1, "Machine Name": "zeroconf_test"},
+    }
+    result = await hass.config_entries.flow.async_init(
+        DOMAIN, context={"source": SOURCE_ZEROCONF}, data=discovery_info
+    )
+    await hass.async_block_till_done()
+    assert result["type"] == data_entry_flow.RESULT_TYPE_ABORT
+    assert config_entry.title == "zeroconf_test"
+    assert len(hass.config_entries.async_entries(DOMAIN)) == 2
+
+
+async def test_config_flow_no_websocket(hass, config_entry):
+    """Test config flow setup without websocket enabled on server."""
+    with patch(
+        "homeassistant.components.forked_daapd.config_flow.ForkedDaapdAPI.test_connection"
+    ) as mock_test_connection:
+        # test invalid config data
+        mock_test_connection.return_value = ["websocket_not_enabled"]
+        result = await hass.config_entries.flow.async_init(
+            DOMAIN, context={"source": SOURCE_USER}, data=config_entry.data
+        )
+        assert result["type"] == data_entry_flow.RESULT_TYPE_FORM
+
+
+async def test_config_flow_zeroconf_invalid(hass):
+    """Test that an invalid zeroconf entry doesn't work."""
+    discovery_info = {"host": "127.0.0.1", "port": 23}
+    result = await hass.config_entries.flow.async_init(
+        DOMAIN, context={"source": SOURCE_ZEROCONF}, data=discovery_info
+    )  # doesn't create the entry, tries to show form but gets abort
+    assert result["type"] == data_entry_flow.RESULT_TYPE_ABORT
```

Comment:
![Codacy](https://app.codacy.com/assets/images/favicon.png) Issue found: [Use of assert detected. The enclosed code will be removed when compiling to optimised byte code.](https://app.codacy.com/manual/foreign-sub/home-assistant/pullRequest?prid=5518609)